roms_tools/setup/datasets.py

```diff
@@ -23,7 +24,7 @@ class Dataset:
     Parameters
     ----------
     filename : str
-        The path to the data files. Can contain wildcards.
+        The path to the data files. Can contain wildcards if use_dask is True.
```

Comment:
There's a potential trap here - wildcards sort alphabetically, and because we're using `combine='nested'` then the datasets will be concatenated in the order given. So if our datasets are ever stored under paths which shouldn't be concatenated in alphabetical order then this will give an unintended result.

Reply:
Good point. Can you think of a better solution in terms of the kwargs that we give to `xr.open_mfdataset`?

Reply:
1. Use `combine='by_coords'` (only works if there are coordinates to use to combine, but for `time` there should be),
2. Just make users pass a `list[str]` instead of a wildcard. Then there are no surprises, and that list should be passed directly on to `xr.open_mfdataset`. 
 
I think I would prefer option (2) as it feels less likely to cause any surprises.

Reply:
The problem with (2) is that the list may be very long. For example, the GLORYS typically has one file per day. If the user wants to produce input data for a year, they have to make a list that consists of 365 items. @dafyddstephenson do you have thoughts on this?

Reply:
Would you be okay with (1) @TomNicholas? I just head from @dafyddstephenson that he would prefer (1) as well. 

Just to clarify, if we go for (1), we would change
```
                    ds = xr.open_mfdataset(
                        self.filename,
                        combine="nested",
                        concat_dim=self.dim_names["time"],
                        coords="minimal",
                        compat="override",
                        chunks=chunks,
                    )
```
to
```
                    ds = xr.open_mfdataset(
                        self.filename,
                        combine="by_coords",
                        coords="minimal",
                        compat="override",
                        chunks=chunks,
                    )
```
???

Reply:
Yes - but maybe just double-check that the additional checking done by `"by_coords"` doesn't significantly slow down the opening.

Reply:
Good point, now I remember. This performance issue is the reason I went for 
```
                        combine="nested",
                        concat_dim=self.dim_names["time"],
```
in the first place.

If you have 365 GLORYS files sitting in your directory, `combine="by_coords"` slows the process down by more than 40 seconds, while the prior solution is almost instant.

Reply:
We could support both (1) and (2). 

(1) If the user wants to use wildcards, we open with `combine="by_coords"` but the user has to be fine with a significant performance hit.
(2) If the user provides a list, we open with `combine="nested", concat_dim=self.dim_names["time"]`, and it will be faster.

Reply:
It's a little more confusing than just having one option but I think thats fine.